# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: gfx/angle/checkout/src/libANGLE/VaryingPacking.h
# Commit: 3a91885d47e0
# Full Hash: 3a91885d47e0ef7fba9749562d52300eea67c651
# Author: Miko Mynttinen <mikokm@gmail.com>
# Date: 2019-11-11 21:52:52
# Regressor Bug: 1578576
# File Overlap Count: 4
# Description:
#   Bug 1578576 - Part 2: Update to ANGLE 3865 r=jgilbert
#   
#   Depends on D44561
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D44579
# ==============================================================================

diff -r 77cee77c0307 -r 3a91885d47e0 gfx/angle/checkout/src/libANGLE/VaryingPacking.h
--- a/gfx/angle/checkout/src/libANGLE/VaryingPacking.h	Fri Nov 08 16:41:42 2019 +0000
+++ b/gfx/angle/checkout/src/libANGLE/VaryingPacking.h	Fri Nov 08 17:06:00 2019 +0000
@@ -29,16 +29,18 @@
 struct PackedVarying
 {
     PackedVarying(const sh::ShaderVariable &varyingIn, sh::InterpolationType interpolationIn)
-        : PackedVarying(varyingIn, interpolationIn, "")
+        : PackedVarying(varyingIn, interpolationIn, "", false)
     {}
     PackedVarying(const sh::ShaderVariable &varyingIn,
                   sh::InterpolationType interpolationIn,
-                  const std::string &parentStructNameIn)
+                  const std::string &parentStructNameIn,
+                  GLuint fieldIndexIn)
         : varying(&varyingIn),
           vertexOnly(false),
           interpolation(interpolationIn),
           parentStructName(parentStructNameIn),
-          arrayIndex(GL_INVALID_INDEX)
+          arrayIndex(GL_INVALID_INDEX),
+          fieldIndex(fieldIndexIn)
     {}
 
     bool isStructField() const { return !parentStructName.empty(); }
@@ -73,6 +75,10 @@
     std::string parentStructName;
 
     GLuint arrayIndex;
+
+    // Field index in the struct.  In Vulkan, this is used to assign a
+    // struct-typed varying location to the location of its first field.
+    GLuint fieldIndex;
 };
 
 struct PackedVaryingRegister final
@@ -171,7 +177,10 @@
         return static_cast<unsigned int>(mRegisterList.size());
     }
 
-    const std::vector<std::string> &getInactiveVaryingNames() const;
+    const std::vector<std::string> &getInactiveVaryingNames() const
+    {
+        return mInactiveVaryingNames;
+    }
 
   private:
     bool packVarying(const PackedVarying &packedVarying);